fastly/backend.go

```diff
@@ -96,36 +96,36 @@ type CreateBackendInput struct {
 	// ServiceVersion is the specific configuration version (required).
 	ServiceVersion int
 
-	Name                string      `url:"name,omitempty"`
-	Comment             string      `url:"comment,omitempty"`
 	Address             string      `url:"address,omitempty"`
-	Port                *uint       `url:"port,omitempty"`
-	OverrideHost        string      `url:"override_host,omitempty"`
+	AutoLoadbalance     Compatibool `url:"auto_loadbalance,omitempty"`
+	BetweenBytesTimeout *uint       `url:"between_bytes_timeout,omitempty"`
+	Comment             string      `url:"comment,omitempty"`
 	ConnectTimeout      *uint       `url:"connect_timeout,omitempty"`
-	MaxConn             *uint       `url:"max_conn,omitempty"`
 	ErrorThreshold      *uint       `url:"error_threshold,omitempty"`
 	FirstByteTimeout    *uint       `url:"first_byte_timeout,omitempty"`
-	BetweenBytesTimeout *uint       `url:"between_bytes_timeout,omitempty"`
-	AutoLoadbalance     Compatibool `url:"auto_loadbalance,omitempty"`
-	Weight              *uint       `url:"weight,omitempty"`
-	RequestCondition    string      `url:"request_condition,omitempty"`
 	HealthCheck         string      `url:"healthcheck,omitempty"`
-	Shield              string      `url:"shield,omitempty"`
-	UseSSL              Compatibool `url:"use_ssl,omitempty"`
-	// NOTE: Fastly API sets "ssl_check_cert" to true as its default value
-	// if this parameter is not present in the request.
-	// Removing omitempty from this particular field so that we can still
-	// create a new backend with "ssl_check_cert: false" set.
-	SSLCheckCert    Compatibool `url:"ssl_check_cert"`
-	SSLCACert       string      `url:"ssl_ca_cert,omitempty"`
-	SSLClientCert   string      `url:"ssl_client_cert,omitempty"`
-	SSLClientKey    string      `url:"ssl_client_key,omitempty"`
-	SSLHostname     string      `url:"ssl_hostname,omitempty"`
-	SSLCertHostname string      `url:"ssl_cert_hostname,omitempty"`
-	SSLSNIHostname  string      `url:"ssl_sni_hostname,omitempty"`
-	MinTLSVersion   string      `url:"min_tls_version,omitempty"`
-	MaxTLSVersion   string      `url:"max_tls_version,omitempty"`
-	SSLCiphers      string      `url:"ssl_ciphers,omitempty"`
+	MaxConn             *uint       `url:"max_conn,omitempty"`
+	MaxTLSVersion       string      `url:"max_tls_version,omitempty"`
+	MinTLSVersion       string      `url:"min_tls_version,omitempty"`
+	Name                string      `url:"name,omitempty"`
+	OverrideHost        string      `url:"override_host,omitempty"`
+	Port                uint        `url:"port,omitempty"`
```

Comment:
ℹ️ I changed the `Port` field from a pointer to primitive as I don't believe anyone would want to set the port number to zero intentionally and so having the zero value omitted in this case should be fine.